cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -17,79 +17,135 @@ package kafkareceiver
 import (
 	"context"
 
-	"github.com/uber/jaeger-lib/metrics"
+	"github.com/Shopify/sarama"
+	"github.com/spf13/viper"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/configtls"
 	"go.opentelemetry.io/collector/consumer"
-	"go.opentelemetry.io/collector/obsreport"
-	jaegertranslator "go.opentelemetry.io/collector/translator/trace/jaeger"
-	"go.uber.org/zap"
-
-	"github.com/jaegertracing/jaeger/cmd/ingester/app/builder"
-	ingester "github.com/jaegertracing/jaeger/cmd/ingester/app/consumer"
-	"github.com/jaegertracing/jaeger/model"
-	"github.com/jaegertracing/jaeger/storage/spanstore"
-)
+	"go.opentelemetry.io/collector/exporter/kafkaexporter"
+	"go.opentelemetry.io/collector/receiver/kafkareceiver"
 
-var (
-	_ spanstore.Writer   = (*writer)(nil)
-	_ component.Receiver = (*kafkaReceiver)(nil)
+	ingesterApp "github.com/jaegertracing/jaeger/cmd/ingester/app"
+	"github.com/jaegertracing/jaeger/plugin/storage/kafka"
 )
 
-type kafkaReceiver struct {
-	logger   *zap.Logger
-	consumer *ingester.Consumer
+// TypeStr defines receiver type.
+const TypeStr = "kafka"
+
+// Factory wraps kafkareceiver.Factory and makes the default config configurable via viper.
+// For instance this enables using flags as default values in the config object.
+type Factory struct {
+	// Wrapped is Kafka receiver.
+	Wrapped component.ReceiverFactory
+	// Viper is used to get configuration values for default configuration
+	Viper *viper.Viper
 }
 
-type writer struct {
-	receiver     string
-	nextConsumer consumer.TraceConsumer
+var _ component.ReceiverFactory = (*Factory)(nil)
+
+// Type returns the type of the receiver.
+func (f *Factory) Type() configmodels.Type {
+	return f.Wrapped.Type()
 }
 
-func new(
-	config *Config,
-	nextConsumer consumer.TraceConsumer,
+// CreateDefaultConfig returns default configuration of Factory.
+// This function implements OTEL component.ReceiverFactoryBase interface.
+func (f *Factory) CreateDefaultConfig() configmodels.Receiver {
+	cfg := f.Wrapped.CreateDefaultConfig().(*kafkareceiver.Config)
+	// load jaeger config
+	opts := &ingesterApp.Options{}
+	opts.InitFromViper(f.Viper)
+
+	cfg.Brokers = opts.Brokers
+	cfg.ClientID = opts.ClientID
+	cfg.Encoding = MustOtelEncodingForJaegerEncoding(opts.Encoding)
+	cfg.GroupID = opts.GroupID
+	cfg.Topic = opts.Topic
+	cfg.ProtocolVersion = opts.ProtocolVersion
+
+	// kafka consumer groups require a min version of V0_10_2_0.  if no version is specified
+	//  we will assume this
+	if len(cfg.ProtocolVersion) == 0 {
+		cfg.ProtocolVersion = sarama.V0_10_2_0.String()
```

Comment:
Shouldn't this be set as a flag default value?

Reply:
Right now, Jaeger has no default which seems to work fine.  However, when using the OTEL Jaeger receiver it fails with the following error when empty:

```
Failed to run the service: cannot setup pipelines: cannot build receivers: cannot create receiver kafka: kafka: invalid configuration (consumer groups require Version to be >= V0_10_2_0)
```

I could make the Jaeger default "0.10.2.0" if desired, but it made more sense to me to force it here if unspecified.

Reply:
sounds good we can set the default here.